City-of-Helsinki/drupal-hdbt
# UHF-3369 Node version update

How to test:
- Would be best to test this branch in a Drupal instance, like sote or kymp
- Start by removing `node_modules` folder: `rm -rf node_modules`
- Checkout this branch or run `composer require drupal/hdbt:dev-UHF-3369_node_version_update` depending your approach of testing this PR
- Once correct branch is checked out, run `nvm use && npm i`
   - Check that there are no critical security issues (should be no issues actually)
- Run `npm run dev` or `npm run build`
   - Check that SCSS and JS gets compiled like before
- If you're testing this change in a Drupal instance, log in and Flush all caches from the UI. Wait until done and explore the site and look for broken styles. I compared the 
   - `dist/css/styles.min.css` compiled with Node 14 and Node Sass and 
   - `dist/css/styles.min.css` compiled with Node 16 and Dart Sass 
   with a diff tool (Kaleidoscope), and there were only minor changes like `.5em` values were changed to `0.5em`,  single quotes changed to double quotes, colors were changed to lower case `#4169AF` -> `#4169af`, etc. 
   
Node version was also updated in these repositories. Please check them as well.
- https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/77/files
- https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/236/files
- https://github.com/City-of-Helsinki/drupal-helfi-platform/pull/50/files
